### ISSUE TITLE
[studio-ui] Shared component path is not updated in parent renamed

### ISSUE THREAD

**Issue by AndyReiter** [craftercms/craftercms#6150]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the latest released 4.2.x
- [X] The issue is in the latest released 4.1.x
- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

When editing the Page URL of an embedded piece of content, the parent content does not get updated with the new Page URL. This occurs for both Page URLs as well as Component ID's.

### Steps to reproduce

Steps:

1. Create a site based on Editorial
2. Edit the home page
3. Scroll down and edit the Left Rails Component that is referenced and edit it
4. Edit the Component ID and add a `-test` at the end
5. Save & Close the component's form. 
   - Notice even immediately trying to edit that same component again shows a workflow cancellation dialog that spins eternally. This is because of the underlying 404. Please fix Workflow cancellation dialog error handling too. Should do this first to test while the rename bug is present.
6. Save & Close the home page
7. Edit the Home Page
8. Notice the Left Rails Component does not have the `-test` at the end
9. Attempt to Edit the Component and see the error

### Relevant log output

```shell
https://gist.github.com/AndyReiter/e9ee5c33dfb7fb4adedb2a2a08891976
```

**Comment by russdanner** [comment]
Scenario:
- A content item P has an item selector that points to another content object Q
- The editor opens A and then using the item selector opens Q.
- In Q they change the file-name of the content and save the form. Taking them back to P.
The expectation is that the item selector would now show the new value for Q's path.  It does not.

**Comment by rart** [comment]
@jvega190 also, let's change up that warning dialog to something similar to the asset rename dialog. Ping me for details.

**Comment by jvega190** [comment]
@AndyReiter This can be tested now in 4.1.x. PR pending to merge in 4.0.x

**Comment by AndyReiter** [comment]
Confirmed fixed in the latest build of 4.1.x.
